ibis/pyspark/compiler.py

```diff
@@ -497,6 +507,234 @@ def compile_isinf(t, expr, scope, **kwargs):
     return (src_column == float('inf')) | (src_column == float('-inf'))
 
 
+@compiles(ops.Uppercase)
+def compile_uppercase(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.upper(src_column)
+
+
+@compiles(ops.Lowercase)
+def compile_lowercase(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.lower(src_column)
+
+
+@compiles(ops.Reverse)
+def compile_reverse(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.reverse(src_column)
+
+
+@compiles(ops.Strip)
+def compile_strip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.trim(src_column)
+
+
+@compiles(ops.LStrip)
+def compile_lstrip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.ltrim(src_column)
+
+
+@compiles(ops.RStrip)
+def compile_rstrip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.rtrim(src_column)
+
+
+@compiles(ops.Capitalize)
+def compile_capitalize(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.initcap(src_column)
+
+
+@compiles(ops.Substring)
+def compile_substring(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @F.udf('string')
```

Comment:
Do pandas_udfs support string types? There is `pandas.Series.str.slice`.

Reply:
Yes, but start and stop are scalars for pandas `str.slice`. In this case, they can be columns.